### DISCUSSION TITLE
New transform Mod Setting (Emit)

### DISCUSSION THREAD

**Discussion by mk56-spn** [ppy/osu#19446] [Ideas]
I had this idea i was experimenting with of having a fixed point emit all the notes to their original places, somewhat like mobile mania games that have a semi circle from which the notes spread out. here i decided to center it in the middle of the playfield but really i could make settings for the mod allowing it to be placed anywhere if that's wanted, making this a discussion so i can gaige player and dev interest in having this in Lazer before i make tests and the like, either way here is the video:

the emission speed is locked to the AR because i found that to be the most comfortable
High ar: 
https://streamable.com/jly6zr
Low ar:
https://streamable.com/von8hx

**Comment by AadilRn** [comment]
I feel as if this is similar to the transform mod, the only difference being the origin point is random for the transform mod (as far as I know, I have not looked at the transform mod algorithm). Maybe instead of a new mod this could be a setting for transform.

If this does become a standalone mod, what do you think of getting rid of the approach circle and lining up the hit timing of the note with the time it takes to get to its correct position?

**Reply by peppy** [reply]
A setting for transform mod does sound like a good place to house this, if it works from a code perspective.

**Reply by mk56-spn** [reply]
I'll check it out. I based my code on wiggle though so I might have to redo it. As a sidenote . Do mods currently support having settings that only show once another setting is toggled? I can't think of a case right now of nested settings

**Reply by AadilRn** [reply]
What would you need nested settings for? I think you only need 1 setting that chooses a specific place on the playfield where the notes will emit from.

**Reply by mk56-spn** [reply]
As I stated above I wanted to make the place of emission user selectable

**Reply by AadilRn** [reply]
Couldn't that just be a drop down menu similar to menus like in approach different? Or I'm not sure what way you are trying to implement that as a setting.

**Reply by mk56-spn** [reply]
> Couldn't that just be a drop down menu similar to menus like in approach different? Or I'm not sure what way you are trying to implement that as a setting.

Two bars like the tablet offset subsetting to select the place of origin on the playfield, by the way with this added the tooltop for transform becomes rather not adequate, any suggestions for what to change it to?

**Reply by mk56-spn** [reply]
https://streamable.com/jp7whu refactored it into transform without any settings for now
